configuration/src/main/kotlin/com/malinskiy/marathon/config/serialization/ConfigurationFactory.kt

```diff
@@ -106,10 +107,13 @@ class ConfigurationFactory(
                 VendorConfiguration.StubVendorConfiguration -> configuration.vendorConfiguration
                 is VendorConfiguration.EmptyVendorConfiguration -> throw ConfigurationException("No vendor configuration specified")
             }
-
             return configuration.copy(
                 vendorConfiguration = vendorConfiguration,
-                analyticsTracking = analyticsTracking ?: configuration.analyticsTracking
+                // Default value for analyticsTracking / bugsnagReporting in both CLI and marathon / gradle config is true.
+                // If analyticsTracking / bugsnagReporting is set to false in either CLI or marathon / gradle config, it will be disabled.
+                // Note that it's not possible to set the CLI value when running marathon via gradle.
+                analyticsTracking = !(analyticsTracking == false || configuration.analyticsTracking == false),
```

Comment:
To be honest, such constructions are a bit confusing. For me, it is better to write a bot more but make the statement clear. For example,
```
analyticsTracking = 
    if (analyticsTracking == false || configuration.analyticsTracking == false) false 
    then true
```
But this comment may be subjective.

Reply:
@tabletenniser thank you for your contribution! Approved.
I've added a small comment. Please have a look.